File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/application/user/command/RegisterUserService.java
package dev.starryeye.book_library.application.user.command;

import dev.starryeye.book_library.application.user.command.command.RegisterUserCommand;
import dev.starryeye.book_library.application.user.command.result.RegisterUserResult;
import dev.starryeye.book_library.domain.user.User;
import dev.starryeye.book_library.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional
@Service
@RequiredArgsConstructor
public class RegisterUserService {

    private final UserRepository userRepository;

    public RegisterUserResult register(RegisterUserCommand command) {

        User user = User.createUser(
                command.username(),
                command.age()
        );

        userRepository.save(user);

        return RegisterUserResult.of(user);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/command/RegisterUserService.kt
package dev.starryeye.book_library.application.user.command

import dev.starryeye.book_library.application.user.command.command.RegisterUserCommand
import dev.starryeye.book_library.application.user.command.result.RegisterUserResult
import dev.starryeye.book_library.domain.user.User
import dev.starryeye.book_library.domain.user.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class RegisterUserService(
    private val userRepository: UserRepository,
) {

    fun register(command: RegisterUserCommand): RegisterUserResult {

        val user = User(
            username = command.username,
            age = command.age,
        )

        userRepository.save(user)

        return RegisterUserResult.of(user)
    }
}